### PR TITLE
chore: refactor packages structure

### PR DESCRIPTION
**Pull request by gsanchezgavier**
- Moves the v3 metadata abstraction to a plain one since versions are compatible and they will remain compatible in the future: [docs](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task-metadata-endpoint-v4.html) ,[code](https://github.com/aws/amazon-ecs-agent/blob/174ace6d165eaa4bc8f8f4be0106519c818d0b07/agent/handlers/v4/response.go#L30)
- Moves things from pkg to internal
- Change `DEBUG_MODE` behavior to print the metadata/task json output to the logs in case this is needed for troubleshooting.

### CODE REVIEW

File: cmd/nri-ecs.go [hunk 1]
```diff
@@ -34,91 +34,59 @@ func main() {
 		os.Exit(1)
 	}
 
-	if args.DebugMode {
-		v3.DebugECSEndpoint()
```
Comment (paologallinaharbur) [inline cmd/nri-ecs.go:38]: Do we know why this was added? Is it possible that in some scenario is still useful when triaging issues? 
maybe @alejandrodnm remembers something 😄

File: cmd/nri-ecs.go [hunk 2]
```diff
@@ -34,91 +34,59 @@ func main() {
...
+		log.Error(
 			"unable to get response from v3 task metadata endpoint (%s): %v",
 			taskMetadataEnpoint,
 			err,
 		)
 		os.Exit(1)
 	}
 
-	var taskMetadata v3.TaskResponse
+	if args.DebugMode {
```
Comment (paologallinaharbur) [inline cmd/nri-ecs.go:55]: shouldn't we set the debug level if args.verbose and change this to log.Debug?

File: cmd/nri-ecs.go [hunk 3]
```diff
@@ -34,91 +34,59 @@ func main() {
...
-		ecsIntegration.Logger().Errorf("unable to find task metadata endpoint")
+		log.Error("unable to find task metadata endpoint")
 		os.Exit(1)
 	}
 
-	body, err := v3.MetadataResponse(httpClient, taskMetadataEnpoint)
+	body, err := metadata.MetadataResponse(httpClient, taskMetadataEnpoint)
 	if err != nil {
-		ecsIntegration.Logger().Errorf(
+		log.Error(
```
Comment (paologallinaharbur) [inline cmd/nri-ecs.go:47]: log.Fatal removing the os.Exit(1)?

File: cmd/nri-ecs.go [hunk 4]
```diff
@@ -34,91 +34,59 @@ func main() {
...
-		os.Exit(0)
-	}
-
 	httpClient := ecs.ClientWithTimeout(5 * time.Second)
 
-	taskMetadataEnpoint, found := v3.TaskMetadataEndpoint()
+	taskMetadataEnpoint, found := metadata.TaskMetadataEndpoint()
 	if !found {
-		ecsIntegration.Logger().Errorf("unable to find task metadata endpoint")
+		log.Error("unable to find task metadata endpoint")
```
Comment (paologallinaharbur) [inline cmd/nri-ecs.go:41]: same here, log.Fatal?

**Review by paologallinaharbur — APPROVED** [review]
Some small things regarding logging.
Before merging I would try to contact FSI to ask if that debug mode was there for some reason

File: cmd/nri-ecs.go (hunk 1 above)
Comment (alejandrodnm) [reply cmd/nri-ecs.go:38]: That's just something to check the endpoints on startup, I think we copied it from an official AWS ECS client.

File: cmd/nri-ecs.go (hunk 2 above)
Comment (gsanchezgavier) [reply cmd/nri-ecs.go:55]: yes that make sense!

**Pull request merged**